src/test/java/org/fogbowcloud/manager/core/ApplicationFacadeTest.java

```diff
@@ -12,1701 +12,2209 @@
-
-		DatabaseManager databaseManager = Mockito.mock(DatabaseManager.class);
-		when(databaseManager.readActiveOrders(OrderState.OPEN)).thenReturn(new SynchronizedDoublyLinkedList());
-		when(databaseManager.readActiveOrders(OrderState.SPAWNING)).thenReturn(new SynchronizedDoublyLinkedList());
-		when(databaseManager.readActiveOrders(OrderState.FAILED)).thenReturn(new SynchronizedDoublyLinkedList());
-		when(databaseManager.readActiveOrders(OrderState.FULFILLED)).thenReturn(new SynchronizedDoublyLinkedList());
-		when(databaseManager.readActiveOrders(OrderState.PENDING)).thenReturn(new SynchronizedDoublyLinkedList());
-		when(databaseManager.readActiveOrders(OrderState.CLOSED)).thenReturn(new SynchronizedDoublyLinkedList());
-
-		doNothing().when(databaseManager).add(any(Order.class));
-		doNothing().when(databaseManager).update(any(Order.class));
-
-		PowerMockito.mockStatic(DatabaseManager.class);
-		given(DatabaseManager.getInstance()).willReturn(databaseManager);
-		
-		this.orderController = Mockito.spy(new OrderController());
-		this.application = Mockito.spy(ApplicationFacade.getInstance());
-		this.application.setAaController(this.aaaController);
-		this.application.setOrderController(this.orderController);
-
-		SharedOrderHolders sharedOrderHolders = SharedOrderHolders.getInstance();
-		this.activeOrdersMap = sharedOrderHolders.getActiveOrdersMap();
-	}
+    @Before
+    public void setUp() throws UnauthorizedRequestException {
+        this.aaaController = Mockito.mock(AaController.class);
+
+        HomeDir.getInstance().setPath("src/test/resources/private");
+
+        DatabaseManager databaseManager = Mockito.mock(DatabaseManager.class);
+        Mockito.when(databaseManager.readActiveOrders(OrderState.OPEN))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.SPAWNING))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.FAILED))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.FULFILLED))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.PENDING))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+        Mockito.when(databaseManager.readActiveOrders(OrderState.CLOSED))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+
+        Mockito.doNothing().when(databaseManager).add(Matchers.any(Order.class));
+        Mockito.doNothing().when(databaseManager).update(Matchers.any(Order.class));
+
+        PowerMockito.mockStatic(DatabaseManager.class);
+        BDDMockito.given(DatabaseManager.getInstance()).willReturn(databaseManager);
+
+        this.orderController = Mockito.spy(new OrderController());
+        this.application = Mockito.spy(ApplicationFacade.getInstance());
```

Comment:
Please, check if it we really need to use a spies. If not, avoid using them.

Reply:
The spy on line 76 can be removed without affecting the code.
however, at line 75 its removal would cause failure in 12 of the 81 tests, and then I'll keep it.